File: server/router/system/register_task.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type RegisterTaskRouter struct{}

func (r *RegisterTaskRouter) InitRegisterTaskRouter(Router *gin.RouterGroup) {
	registerTaskRouter := Router.Group("registerTask").Use(middleware.OperationRecord())
	registerTaskRouterWithoutRecord := Router.Group("registerTask")
	{
		registerTaskRouter.POST("create", registerTaskApi.CreateRegisterTask) // 地推创建任务
		registerTaskRouter.POST("step", registerTaskApi.SubmitRegisterTaskStep)
		registerTaskRouter.POST("debug/login/start", registerTaskApi.StartRegisterTaskDebugLogin)
		registerTaskRouter.POST("debug/login/submit", registerTaskApi.SubmitRegisterTaskDebugLoginCode)
	}
	{
		registerTaskRouterWithoutRecord.GET("active", registerTaskApi.GetActiveRegisterTask)
		registerTaskRouterWithoutRecord.GET("actives", registerTaskApi.GetActiveRegisterTasks)
		registerTaskRouterWithoutRecord.POST("list", registerTaskApi.GetRegisterTaskList)
		registerTaskRouterWithoutRecord.GET("summary", registerTaskApi.GetRegisterTaskSummary)
		registerTaskRouterWithoutRecord.GET("debug/login/task", registerTaskApi.GetRegisterTaskDebugLoginTask)
		registerTaskRouterWithoutRecord.GET("cache/download", registerTaskApi.DownloadRegisterTaskCache)
	}
}

File: web/src/api/registerTask.js
import service from '@/utils/request'

export const createRegisterTask = (data) => {
  return service({
    url: '/registerTask/create',
    method: 'post',
    data
  })
}

export const getActiveRegisterTask = (config = {}) => {
  return service({
    url: '/registerTask/active',
    method: 'get',
    ...config
  })
}

export const getActiveRegisterTasks = (config = {}) => {
  return service({
    url: '/registerTask/actives',
    method: 'get',
    ...config
  })
}

export const submitRegisterTaskStep = (data) => {
  return service({
    url: '/registerTask/step',
    method: 'post',
    data
  })
}

export const getRegisterTaskList = (data, config = {}) => {
  return service({
    url: '/registerTask/list',
    method: 'post',
    data,
    ...config
  })
}

export const getRegisterTaskSummary = (params) => {
  return service({
    url: '/registerTask/summary',
    method: 'get',
    params
  })
}

export const settleRegisterTaskLeader = (data) => {
  return service({
    url: '/registerTask/settle',
    method: 'post',
    data
  })
}

export const getRegisterTaskSettlementHistory = (params) => {
  return service({
    url: '/registerTask/settlement/history',
    method: 'get',
    params
  })
}

export const startRegisterTaskDebugLogin = (data) => {
  return service({
    url: '/registerTask/debug/login/start',
    method: 'post',
    data
  })
}

export const submitRegisterTaskDebugLoginCode = (data) => {
  return service({
    url: '/registerTask/debug/login/submit',
    method: 'post',
    data
  })
}

export const getRegisterTaskDebugLoginTask = (params) => {
  return service({
    url: '/registerTask/debug/login/task',
    method: 'get',
    params
  })
}

export const downloadRegisterTaskCache = (params) => {
  return service({
    url: '/registerTask/cache/download',
    method: 'get',
    params,
    responseType: 'arraybuffer',
    donNotShowLoading: true
  })
}
